CoreScriptsRoot/Modules/ChatSelector.lua

```diff
@@ -0,0 +1,162 @@
+local GuiService = game:GetService("GuiService")
+
+local Util = require(RobloxGui.Modules.ChatUtil)
+
+local function GetUseLuaFlag()
+	local loop_continue = true
+	while loop_continue do
+		local success, retVal = pcall(function()
+			return game.IsSFFlagsLoaded
+		end)
+		if not success then
+			loop_continue = false
+		elseif retVal then
+			loop_continue = false
+		else
+			wait(0.1)
+		end
+	end
+
+	local success, retVal = pcall(function() return game:GetService("Chat"):GetShouldUseLuaChat() end)
+	local useNewChat = success and retVal
+	return useNewChat
+end
+
+local readFlagSuccess, flagEnabled = pcall(function() return settings():GetFFlag("CorescriptNewLoadChat") end)
+local TryLoadNewChat = readFlagSuccess and flagEnabled
+
+
+local useModule = nil
+
+local state = {Visible = true}
+local interface = {}
+do
+	function interface:ToggleVisibility()
+		if (useModule) then
+			useModule:ToggleVisibility()
+		else
+			state.Visible = not state.Visible
+		end
+	end
+
+	function interface:SetVisible(visible)
+		if (useModule) then
+			useModule:SetVisible(visible)
+		else
+			state.Visible = visible
+		end
+	end
+
+	function interface:FocusChatBar()
+		if (useModule) then
+			useModule:FocusChatBar()
+		else
+			--// do nothing
+		end
+	end
+
+	function interface:GetVisibility()
+		if (useModule) then
+			return useModule:GetVisibility()
+		else
+			return state.Visible
+		end
+	end
+
+	function interface:GetMessageCount()
+		if (useModule) then
+			return useModule:GetMessageCount()
+		else
+			return 0
+		end
+	end
+
+	function interface:TopbarEnabledChanged(...)
+		if (useModule) then
+			return useModule:TopbarEnabledChanged(...)
+		else
+			
+		end
+	end
+
+	function interface:IsFocused(useWasFocused)
+		if (useModule) then
+			return useModule:IsFocused(useWasFocused)
+		else
+			return false
+		end
+	end
+
+	interface.ChatBarFocusChanged = Util.Signal()
+	interface.VisibilityStateChanged = Util.Signal()
+	interface.MessagesChanged = Util.Signal()
+end
+
+local stopCachingMakeSystemMessage = false
+local MakeSystemMessageCache = {}
+local function MakeSystemMessageCachingFunction(data)
+	if (stopCachingMakeSystemMessage) then return end
+	table.insert(MakeSystemMessageCache, data)
+end
+
+local function NonFunc() end
+StarterGui:RegisterSetCore("ChatMakeSystemMessage", MakeSystemMessageCachingFunction)
+StarterGui:RegisterSetCore("ChatWindowPosition", NonFunc)
+StarterGui:RegisterSetCore("ChatWindowSize", NonFunc)
+StarterGui:RegisterGetCore("ChatWindowPosition", NonFunc)
+StarterGui:RegisterGetCore("ChatWindowSize", NonFunc)
+StarterGui:RegisterSetCore("ChatBarDisabled", NonFunc)
+StarterGui:RegisterGetCore("ChatBarDisabled", NonFunc)
+
+
+local function ConnectSignals(useModule, interface, sigName)
+	useModule[sigName]:connect(function(...) interface[sigName]:fire(...) end)
+end
+
+local isConsole = GuiService:IsTenFootInterface() 		or false -- set true to force a true value
```

Comment:
Lets get rid of these crazy spaces and the `or false`. If we want to have a way to force it on we could put a flag variable at the top